zproject/backends.py

```diff
@@ -632,13 +632,32 @@ def is_account_control_disabled_user(self, ldap_user: _LDAPUser) -> bool:
         return ldap_disabled
 
     def is_account_realm_access_forbidden(self, ldap_user: _LDAPUser, realm: Realm) -> bool:
-        if "org_membership" not in settings.AUTH_LDAP_USER_ATTR_MAP:
+        if "org_membership" in settings.AUTH_LDAP_USER_ATTR_MAP:
+            org_membership_attr = settings.AUTH_LDAP_USER_ATTR_MAP["org_membership"]
+            allowed_orgs: List[str] = ldap_user.attrs.get(org_membership_attr, [])
+            return not is_subdomain_in_allowed_subdomains_list(realm.subdomain, allowed_orgs)
+        if settings.AUTH_LDAP_ADVANCED_REALM_ACCESS_CONTROL is not None:
+            realm_access_control = settings.AUTH_LDAP_ADVANCED_REALM_ACCESS_CONTROL
+            if (
+                isinstance(realm_access_control, dict)
+                and realm.subdomain in realm_access_control
+                and isinstance(realm_access_control[realm.subdomain], dict)
+            ):
+                for attribute in realm_access_control[realm.subdomain]:
+                    if (
+                        attribute in ldap_user.attrs
+                        and realm_access_control[realm.subdomain][attribute]
+                        in ldap_user.attrs[attribute]
+                    ):
+                        return False
+                # If no match is found forbid access
+                return True
+            else:
+                # If configuration is wrong do not allow access
+                return True
+        else:
```

Comment:
This doesn't need an `else` -- you can just `return False` after all the `if` blocks.

Reply:
I think it would not work here using just return `False` at the end of the method, because we return `True` in Line 657.

Because if at the end of the Method it checks if there are no settings set for `org_membership` or `ADVANCED_REALM_ACCESS_CONTROL` it does allow access.
In line 657 if a wrong configuration is made (e.g. not configuring a realm) we play safe and rather do not allow the access.